src/common/snippets/include/snippets/pass_manager.hpp

```diff
@@ -0,0 +1,100 @@
+// Copyright (C) 2018-2023 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#pragma once
+#include "openvino/pass/manager.hpp"
+#include "openvino/pass/pass.hpp"
+#include "openvino/pass/validate.hpp"
+#include <typeinfo>
+
+
+namespace ov {
+namespace snippets {
+namespace pass {
+/**
+ * @brief Manager is like ov::pass::Manager, but allows to insert new passes at arbitrary places in the pipeline
+ * @ingroup snippets
+ */
+class Manager : public ov::pass::Manager {
+public:
+    ~Manager() override = default;
+    using PassBase = ov::pass::PassBase;
+    using Validate = ov::pass::Validate;
+    /**
+    * @brief PassPosition describes a particular position in a transformation pipeline,
+     * where a new transformation should be inserted.
+     * @param pass_name name of the anchor pass, the new pass will be inserted before/after it.
+     * Empty pass_name could mean either beginning or the end of the pipeline depending on the `after` flag.
+     * No default value.
+     * @param after `true` if the new pass should be inserted before the anchor pass, `false` otherwise (default).
```

Comment:
I came up with an idea that could probably alleviate a little the problems that arise in the situation described by Alexandra. What if we introduce some special pass, whose the main goal is to mark the insertion position in the pipeline? Let's call it `InsertionPositionPlace`.
`InsertionPositionPlace` should do nothing with the model. So we can not insert Validate pass after it. This pass could have string parameter which gives the name to the insertion position.

The common pipeline in this case could seem smth like
```
common_pipeline.register_pass<InsertionPositionPlace>("PipelineBegin");
common_pipeline.register_pass<PassA>();
common_pipeline.register_pass<InsertionPositionPlace>("InsertionPosForBackendB");
common_pipeline.register_pass<PassBeforeA>();
common_pipeline.register_pass<PassA>();
common_pipeline.register_pass<InsertionPositionPlace>("InsertionPosForBackendA");
...
common_pipeline.register_pass<InsertionPositionPlace>("PipelineEnd");
```
After first instance of `PassA` removal:
```
common_pipeline.register_pass<InsertionPositionPlace>("PipelineBegin");
common_pipeline.register_pass<InsertionPositionPlace>("InsertionPosForBackendB");
common_pipeline.register_pass<PassBeforeA>();
common_pipeline.register_pass<PassA>();
common_pipeline.register_pass<InsertionPositionPlace>("InsertionPosForBackendA");
...
common_pipeline.register_pass<InsertionPositionPlace>("PipelineEnd");
```

In the backend, we just specify the name of `InsertionPositionPlace`, where we need to insert backend passes.
As I mentioned, the idea just popped in my head, so I most likely missed something) But if you like the idea, we can discuss it offline

Reply:
Thanks for your ideas 👍 
As discussed offline, it's hardly possible to design a pipeline that wouldn't be easy to break in the common part.
This pass manager is a tool to tweak snippets pipeline. Of course, if we make alter the pipeline significantly, then might need to update the fine-tuning on the backend side. Unfortunately, I don't think there is a way to avoid it.